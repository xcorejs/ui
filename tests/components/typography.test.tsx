import 'jest-styled-components';

import { Box, Typography } from '@xcorejs/ui';
import React from 'react';
import renderer from 'react-test-renderer';

test('Typography component', () => {
  const component = renderer.create(
    <>
      <Box>
        <Typography variant="h1" mb="4rem">Heading 1</Typography>
        <Typography variant="h2">Heading 2</Typography>
        <Typography variant="h3">Heading 3</Typography>
        <Typography variant="h4">Heading 4</Typography>
        <Typography variant="h5">Heading 5</Typography>
        <Typography variant="h6">Heading 6</Typography>
        <Typography variant="p">Paragraph</Typography>
        <Typography variant="lead">Lead</Typography>
        <hr />
        <Typography variant="h2" as="h3">Heading 2 as h3</Typography>
      </Box>
    </>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
