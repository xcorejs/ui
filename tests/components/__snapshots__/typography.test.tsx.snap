// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography component 1`] = `
.c0 {
  margin: 0;
  margin-bottom: 4rem;
  font-size: 4.4rem;
  font-weight: 500;
  font-family: "rubik",-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  line-height: 6.6rem;
  font-weight: 500;
  font-size: 4.4rem;
  color: #1E3441;
}

.c1 {
  margin: 0;
  font-size: 3.2rem;
  font-weight: 500;
  font-family: "rubik",-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  line-height: 4.8rem;
  font-weight: 500;
  font-size: 3.2rem;
  color: #1E3441;
}

.c2 {
  margin: 0;
  font-size: 2.4rem;
  font-weight: 500;
  font-family: "rubik",-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  line-height: 3.6rem;
  font-weight: 500;
  font-size: 2.4rem;
  color: #1E3441;
}

.c3 {
  margin: 0;
  font-family: "rubik",-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  color: #1E3441;
}

.c4 {
  margin: 0;
  font-size: 1.6rem;
  font-family: "rubik",-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  line-height: 2rem;
  font-size: 1.6rem;
  color: #1E3441;
}

<div
  className=""
>
  <h1
    className="c0"
    color="text"
    fontFamily="heading"
    fontSize="4.4rem"
    fontWeight={500}
  >
    Heading 1
  </h1>
  <h2
    className="c1"
    color="text"
    fontFamily="heading"
    fontSize="3.2rem"
    fontWeight={500}
  >
    Heading 2
  </h2>
  <h3
    className="c2"
    color="text"
    fontFamily="heading"
    fontSize="2.4rem"
    fontWeight={500}
  >
    Heading 3
  </h3>
  <h4
    className="c3"
    color="text"
    fontFamily="heading"
  >
    Heading 4
  </h4>
  <h5
    className="c3"
    color="text"
    fontFamily="heading"
  >
    Heading 5
  </h5>
  <h6
    className="c3"
    color="text"
    fontFamily="heading"
  >
    Heading 6
  </h6>
  <p
    className="c4"
    color="text"
    fontFamily="heading"
    fontSize="1.6rem"
  >
    Paragraph
  </p>
  <p
    className="c3"
    color="text"
    fontFamily="heading"
  >
    Lead
  </p>
  <hr />
  <h3
    className="c1"
    color="text"
    fontFamily="heading"
    fontSize="3.2rem"
    fontWeight={500}
  >
    Heading 2 as h3
  </h3>
</div>
`;
